nrich-registry/src/main/java/net/croz/nrich/registry/data/customizer/RegistryDataFormConfigurationMappingCustomizer.java

```diff
@@ -42,21 +44,34 @@ private void registerRegistryFormConfiguration(Map<String, Class<?>> formConfigu
             String registryCreateFormId = String.format(RegistryDataConstants.REGISTRY_FORM_ID_FORMAT, registryClassName, RegistryDataConstants.REGISTRY_FORM_ID_CREATE_SUFFIX);
             String registryUpdateFormId = String.format(RegistryDataConstants.REGISTRY_FORM_ID_FORMAT, registryClassName, RegistryDataConstants.REGISTRY_FORM_ID_UPDATE_SUFFIX);
 
-            if (formConfigurationMap.get(registryCreateFormId) == null) {
-                Class<?> createClass = resolveClass(registryClassName, RegistryDataConstants.CREATE_REQUEST_SUFFIX);
+            formConfigurationMap.computeIfAbsent(registryCreateFormId, key -> resolveClassByPackage(registryClassName, RegistryDataConstants.CREATE_REQUEST_SUFFIX));
+            formConfigurationMap.computeIfAbsent(registryUpdateFormId, key -> resolveClassByPackage(registryClassName, RegistryDataConstants.UPDATE_REQUEST_SUFFIX));
+        });
+    }
 
-                formConfigurationMap.put(registryCreateFormId, createClass);
-            }
-            if (formConfigurationMap.get(registryUpdateFormId) == null) {
-                Class<?> updateClass = resolveClass(registryClassName, RegistryDataConstants.UPDATE_REQUEST_SUFFIX);
+    private Class<?> resolveClassByPackage(String registryClassName, String registrySuffixFormat) {
+        List<String> registryClassNameList = new ArrayList<>(Arrays.asList(registryClassName.split(RegistryDataConstants.PACKAGE_SEPARATOR)));
 
-                formConfigurationMap.put(registryUpdateFormId, updateClass);
-            }
-        });
+        int indexOfLastPackage = registryClassNameList.size() - RegistryDataConstants.LAST_PACKAGE_MARKER_INDEX;
+        if (RegistryDataConstants.SUFFIX_LIST_TO_REMOVE.contains(registryClassNameList.get(indexOfLastPackage))) {
+            registryClassNameList.remove(registryClassNameList.get(indexOfLastPackage));
+        }
+
+        registryClassNameList.add(registryClassNameList.size() - 1, RegistryDataConstants.REQUEST_PACKAGE_NANE);
+
+        String classNameWithRequestPackage = String.join(RegistryCoreConstants.DOT, registryClassNameList);
+
+        return resolveClass(classNameWithRequestPackage, registryClassName, registrySuffixFormat);
     }
 
-    private Class<?> resolveClass(String classFullName, String classLoadingInitialPrefix) {
-        List<String> classNameList = Arrays.asList(String.format(classLoadingInitialPrefix, classFullName), String.format(RegistryDataConstants.REQUEST_SUFFIX, classFullName), classFullName);
+    private Class<?> resolveClass(String classNameWithRequestPackage, String className, String registrySuffixFormat) {
```

Comment:
Maybe rename `className` to `fullClassName` and put it as a first parameter. Additionally, rename `classNameWithRequestPackage` to `fullClassNameWithRequestPackage` and put it as a second parameter.